src/ccl_massfunc.c

```diff
@@ -415,87 +518,101 @@ void ccl_cosmology_compute_sigma(ccl_cosmology * cosmo, int *status)
     strcpy(cosmo->status_message, "ccl_massfunc.c: ccl_cosmology_compute_sigma(): Error creating dlnsigma/dlogM spline\n");
     return;
   }
-  
+
   if(cosmo->data.accelerator_m==NULL)
     cosmo->data.accelerator_m=gsl_interp_accel_alloc();
   cosmo->data.logsigma = logsigma;
   cosmo->data.dlnsigma_dlogm = dlnsigma_dlogm;
   cosmo->computed_sigma = true;
-  
+
   free(m);
   free(y);
 }
 
-/*----- ROUTINE: ccl_massfunc -----
-INPUT: ccl_cosmology * cosmo, double halo mass in units of Msun, double scale factor
-TASK: returns halo mass function as dn / dlog10 m
+/*----- ROUTINE: ccl_dlninvsig_dlogm -----
+INPUT: ccl_cosmology *cosmo, double halo mass in units of Msun
+TASK: returns the value of the derivative of ln(sigma^-1) with respect to log10 in halo mass.
 */
 
-double ccl_massfunc(ccl_cosmology *cosmo, double halomass, double a, double odelta, int * status)
+static double ccl_dlninvsig_dlogm(ccl_cosmology *cosmo, double halomass, int*status)
 {
-  if (cosmo->params.N_nu_mass>0){
-	  *status = CCL_ERROR_NOT_IMPLEMENTED;
-	  strcpy(cosmo->status_message,"ccl_background.c: ccl_cosmology_compute_growth(): Support for the halo mass function in cosmologies with massive neutrinos is not yet implemented.\n");
-	  return NAN; 
-  }
-	
   if (!cosmo->computed_sigma) {
     ccl_cosmology_compute_sigma(cosmo, status);
     ccl_check_status(cosmo, status);
   }
 
-  double f,deriv,rho_m,logmass;
-  
+  double val, logmass;
+
   logmass = log10(halomass);
-  rho_m = RHO_CRITICAL*cosmo->params.Omega_m*cosmo->params.h*cosmo->params.h;
-  f=massfunc_f(cosmo,halomass,a,odelta,status);
-  int gslstatus = gsl_spline_eval_e(cosmo->data.dlnsigma_dlogm, logmass, cosmo->data.accelerator_m,&deriv);
+
+  int gslstatus = gsl_spline_eval_e(cosmo->data.dlnsigma_dlogm, logmass, cosmo->data.accelerator_m,&val);
   if(gslstatus != GSL_SUCCESS) {
     ccl_raise_gsl_warning(gslstatus, "ccl_massfunc.c: ccl_massfunc():");
     *status |= gslstatus;
   }
   ccl_check_status(cosmo, status);
-  return f*rho_m*deriv/halomass;
+
+  return val;
+}
+
+/*----- ROUTINE: ccl_massfunc -----
+INPUT: ccl_cosmology * cosmo, double halo mass in units of Msun, double scale factor
+TASK: returns halo mass function as dn / dlog10 m
```

Comment:
Need to mention units, e.g. is it in comoving or physical Mpc^-3, is it in h^-1 units etc.

Reply:
Done